pkg/apis/stork/v1alpha1/volumeexport.go

```diff
@@ -0,0 +1,118 @@
+	// Rsync supports both local and remote file copy.
+	ExportRsync ExportType = "rsync"
+)
+
+// ExportStatus defines a status of VolumeExport..
+type ExportStatus string
+
+const (
+	// ExportStatusInitial is the initial status of VolumeExport. It indicates
+	// that a volume export request has been received.
+	ExportStatusInitial ExportStatus = "Initial"
+	// ExportStatusPending when data export is pending and not started yet.
+	ExportStatusPending ExportStatus = "Pending"
+	// ExportStatusInProgress when data is being transferred.
+	ExportStatusInProgress ExportStatus = "InProgress"
+	// ExportStatusFailed when data transfer is failed.
+	ExportStatusFailed ExportStatus = "Failed"
+	// ExportStatusSuccessful when data has been transferred.
+	ExportStatusSuccessful ExportStatus = "Successful"
+)
+
+// ExportStage defines different stages for VolumeExport when its Status changes
+// from Initial to Failed/Successful.
+type ExportStage string
+
+const (
+	// ExportStageInitial is the starting point for VolumeExport.
+	ExportStageInitial ExportStage = "Initial"
+	// ExportStageSnapshotScheduled if a driver support this stage, it means a snapshot
+	// is being taken of the source PVC which will be used to transfer data with rsync.
+	ExportStageSnapshotScheduled ExportStage = "SnapshotScheduled"
+	// ExportStageSnapshotInProgress if a driver supports this stage, it means a data
+	// is processing.
+	ExportStageSnapshotInProgress ExportStage = "SnapshotInProgress"
+	// ExportStageTransferScheduled when the rsync daemon and pod are currently being
+	// scheduled by Kubernetes.
+	ExportStageTransferScheduled ExportStage = "TransferScheduled"
+	// ExportStageTransferInProgress when rsync is in progress and is transferring data
+	// between the two PVCs.
+	ExportStageTransferInProgress ExportStage = "TransferInProgress"
+	// ExportStageFinal when rsync is completed.
+	ExportStageFinal ExportStage = "Final"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeExport defines a spec for importing of application data from a non Portworx PVC
+// (source) to a PVC backed by Portworx.
+type VolumeExport struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              VolumeExportSpec   `json:"spec"`
+	Status            VolumeExportStatus `json:"spec"`
+}
+
+// VolumeExportSpec defines configuration parameters for VolumeExport.
+type VolumeExportSpec struct {
+	Type            ExportType              `json:"type,omitempty"`
+	ClusterPairName string                  `json:"clusterPairName,omitempty"`
+	Source          VolumeExportSource      `json:"source,omitempty"`
+	Destination     VolumeExportDestination `json:"destination,omitempty"`
+}
+
+// VolumeExportSource defines a PVC name and namespace that should be processed.
+type VolumeExportSource struct {
+	Name      string `json:"name,omitempty"`
```

Comment:
Similar to the destination, this should have a field for PVC instead of name and namespace here. The source could be something other than PVCs in the future